packages/relay/tests/helpers.ts

```diff
@@ -885,40 +885,70 @@ export const calculateTxRecordChargeAmount = (exchangeRateIncents: number) => {
 };
 
 export const useInMemoryRedisServer = (logger: Logger, port: number) => {
-  let envsToReset: { TEST?: string; REDIS_ENABLED?: string; REDIS_URL?: string };
+  overrideEnvs({ TEST: 'false', REDIS_ENABLED: 'true', REDIS_URL: `redis://127.0.0.1:${port}` });
+
   let redisInMemoryServer: RedisInMemoryServer;
 
   before(async () => {
-    ({ envsToReset, redisInMemoryServer } = await startRedisInMemoryServer(logger, port));
+    redisInMemoryServer = await startRedisInMemoryServer(logger, port);
   });
 
   after(async () => {
-    await stopRedisInMemoryServer(redisInMemoryServer, envsToReset);
+    await stopRedisInMemoryServer(redisInMemoryServer);
   });
 };
 
 export const startRedisInMemoryServer = async (logger: Logger, port: number) => {
   const redisInMemoryServer = new RedisInMemoryServer(logger.child({ name: 'RedisInMemoryServer' }), port);
   await redisInMemoryServer.start();
-  const envsToReset = {
-    TEST: process.env.TEST,
-    REDIS_ENABLED: process.env.REDIS_ENABLED,
-    REDIS_URL: process.env.REDIS_URL,
-  };
-  process.env.TEST = 'false';
-  process.env.REDIS_ENABLED = 'true';
-  process.env.REDIS_URL = `redis://127.0.0.1:${port}`;
-  return { redisInMemoryServer, envsToReset };
+  return redisInMemoryServer;
 };
 
-export const stopRedisInMemoryServer = async (
-  redisInMemoryServer: RedisInMemoryServer,
-  envsToReset: { TEST?: string; REDIS_ENABLED?: string; REDIS_URL?: string },
-): Promise<void> => {
+export const stopRedisInMemoryServer = async (redisInMemoryServer: RedisInMemoryServer): Promise<void> => {
   await redisInMemoryServer.stop();
-  process.env.TEST = envsToReset.TEST;
-  process.env.REDIS_ENABLED = envsToReset.REDIS_ENABLED;
-  process.env.REDIS_URL = envsToReset.REDIS_URL;
+};
+
+export const overrideEnvs = (envs: NodeJS.Dict<string>) => {
+  let envsToReset: NodeJS.Dict<string> = {};
+
+  const overrideEnv = (object: NodeJS.Dict<string>, key: string, value: string | undefined) => {
+    if (value === undefined) {
+      delete object[key];
+    } else {
+      object[key] = value;
+    }
+  };
+
+  before(() => {
+    for (const key in envs) {
+      envsToReset[key] = process.env[key];
+      overrideEnv(process.env, key, envs[key]);
+    }
+  });
+
+  after(() => {
+    for (const key in envs) {
+      overrideEnv(process.env, key, envsToReset[key]);
+    }
+  });
+};
+
+/**
+ * Overrides environment variables for the duration of the provided tests.
+ *
+ * @param {NodeJS.Dict<string>} envs - An object containing key-value pairs of environment variables to set.
+ * @param {Function} tests - A function containing the tests to run with the overridden environment variables.
+ */
+export const withOverriddenEnvs = (envs: NodeJS.Dict<string>, tests: () => void) => {
```

Comment:
Interesting usage.
I see lot sof changes were needed to adopt new approach.
It might be time for a `testing-guide.md` in which there's a section that tells contributors how to approach testing. In one such section you can note these 2 methods and how they should be utilized by engineers.

No need to do the whole doc but at least the how people should use these 2 methods so that contributors are well informed.

Reply:
Hmm, let's open a separate issue for that, I also have a few other ideas of what we can include in this doc

Reply:
https://github.com/hashgraph/hedera-json-rpc-relay/issues/3031